lib/puppet/provider/common.rb

```diff
@@ -121,3 +121,8 @@ def extract_resource_name(class_name)
   shift_prefix = Regexp.last_match.nil? ? 2 : 1
   class_name.split('::')[2].split('_').drop(shift_prefix).collect(&:capitalize).join
 end
+
+def create_image_streamer_client
```

Comment:
This will likely only be used by `lib/puppet/provider/image_streamer_resource.rb`, it should be there.